### ISSUE TITLE
v3.2.2 causes the game to crash to desktop after running sims.exe

### ISSUE THREAD

**Issue by Gaerthal-Estelmist** [FaithBeam/Sims-1-Complete-Collection-Widescreen-Patcher#61]
Hello,
The game worked perfectly with v 3.0.0.
I tried the newly released v 3.2.2 but after running the game shortcut, the screen turns black (as normal) but then the loading screen does not appear and it returns back to the desktop.
I always choose Ddraw compat with 1280x960 resolution. Windows 8.1 64 bit.
I tried with both borderless fullscreen enabled & disabled, same result.

I've been following this project for years now, good job and thank you again for working on it!

**Comment by FaithBeam** [comment]
Hello! This is probably due to the ddrawcompat update to 0.4.0. Can you please patch it again, drop this [ddraw.dll](https://github.com/narzoul/DDrawCompat/releases/download/v0.3.1/DDrawCompat-v0.3.1.zip) in your Sims installation folder, overwrite the existing ddraw.dll, and run The Sims. Let me know if that works. That's the 0.3.1 ddraw.dll that was included in my v3.0.0 release.

You're very welcome! I've not forgotten your photogenic bird profile picture 😁

**Comment by Gaerthal-Estelmist** [comment]
And your kitty always brings a smile :)
I tried your method and it indeed works and the game works normally. Nice!

**Comment by FaithBeam** [comment]
Ok, try this [ddraw](https://github.com/narzoul/DDrawCompat/files/9736501/ddraw.zip). This is Narzoul's fix for someone who was crashing on Windows 7 for 0.4.0. If you crash again, I recommend opening an issue in the [DDrawCompat repository](https://github.com/narzoul/DDrawCompat/issues).

**Comment by Gaerthal-Estelmist** [comment]
> Ok, try this [ddraw](https://github.com/narzoul/DDrawCompat/files/9736501/ddraw.zip). This is Narzoul's fix for someone who was crashing on Windows 7 for 0.4.0. If you crash again, I recommend opening an issue in the [DDrawCompat repository](https://github.com/narzoul/DDrawCompat/issues).

Tried this one, the game did not crash but this time the FPS had a massive drop. The fluidity of Ddraw Compat is there no more. Painful to play.
I guess I'll just stick with v 3.0.0 :)

**Comment by FaithBeam** [comment]
@Gaerthal-Estelmist Can you try the config mrpenguinb wrote in this link and see if that fixes this issue, please? https://github.com/FaithBeam/Sims-1-Complete-Collection-Widescreen-Patcher/issues/64#issuecomment-1449349522

**Comment by Gaerthal-Estelmist** [comment]
> @Gaerthal-Estelmist Can you try the config mrpenguinb wrote in this link and see if that fixes this issue, please? [#64 (comment)](https://github.com/FaithBeam/Sims-1-Complete-Collection-Widescreen-Patcher/issues/64#issuecomment-1449349522)
> 
> Edit: With ddrawcompat v0.4.0.

Hello, sorry I missed your mention. To be honest I had lost hope of getting it to work, so I've not been checking the issues frequently.
After seeing that you updated the release, I came and see your comment here but I guess I'm too late to try it with v0.4.0 since your latest release has Ddrawcompat 0.3.2 . And I can't seem to find how I can try it with 0.4.0 now. Let me know if there is a way to try it now.
For now, I downloaded your latest release (v3.2.3) and it works perfect. No issues. Thanks!

**Comment by PieStars** [comment]
You had to open the DDrawCompat.ini file and enter the line "DisplayResolution = app"

**Comment by PieStars** [comment]
> @Gaerthal-Estelmist Can you try the config mrpenguinb wrote in this link and see if that fixes this issue, please? [#64 (comment)](https://github.com/FaithBeam/Sims-1-Complete-Collection-Widescreen-Patcher/issues/64#issuecomment-1449349522)
> 
> Edit: With ddrawcompat v0.4.0.

You cannot create an experimental version with the line "DisplayResolution = app" already written in .ini. And see if the players will have problems or not?

**Comment by Gaerthal-Estelmist** [comment]
> You had to open the DDrawCompat.ini file and enter the line "DisplayResolution = app"

OK, I installed 3.2.2 and added that line in DDrawCompat.ini but the result is almost the same. I say almost because the only difference before CTD is that the screen becomes full screen first. Before I tried the fix, it used to open black screen in a smaller window. It returns to desktop without loading title sequence.

I then tried with adding these four lines
AltTabFix = keepvidmem
DisplayResolution = app
DisplayRefreshRate = desktop
FpsLimiter = msgloop(60)

But the result was the same again. Returned to desktop. So, no fix for me at the moment and I returned to 3.2.3 .

**Comment by PieStars** [comment]
It's a pity. We hope that in the next update of DDrawcompat this problem will be solved

**Comment by voicemxil** [comment]
> You had to open the DDrawCompat.ini file and enter the line "DisplayResolution = app"
```
AltTabFix = keepvidmem 
DisplayResolution = app 
DisplayRefreshRate = desktop 
FpsLimiter = msgloop(60)
```
> But the result was the same again. Returned to desktop. So, no fix for me at the moment and I returned to 3.2.3 .

@Gaerthal-Estelmist It doesn't look like you've tried the CPU affinity fix yet on 3.2.2. While developing TS1 Starter Pack with the help of Lifen from my Discord, we discovered this tends to resolve the issues coming from DDrawCompat 0.3.2 to 0.4.0, and makes it perform as it did before, if not better.

Instead of `DisplayResolution=app`, can you try putting `CPUAffinity=all` in your config? I've submitted #65 to implement this into the next version of the patcher. 😊

**Comment by Gaerthal-Estelmist** [comment]
Hi voicemxil, reinstalled 3.2.2, updated DDrawCompat.ini as below

FullscreenMode=borderless
CPUAffinity=all 

But the problem remained. Then updated DDrawCompat.ini as below

FullscreenMode=borderless
AltTabFix = keepvidmem 
CPUAffinity=all 
DisplayRefreshRate = desktop 
FpsLimiter = msgloop(60)

Still, the problem remained. Did not go ahead and try with Borderless Fullscreen disabled.

Reverted back to 3.2.3 .

**Comment by voicemxil** [comment]
@Gaerthal-Estelmist Sorry to hear that didn't work for you, it seems to help a lot for others. Would you be willing to share some more of your PC specs? I am willing to look into this further and possibly investigate the source code differences between 0.3.2 and 0.4.0.

**Comment by Gaerthal-Estelmist** [comment]
Hey, this is my laptop: https://www.msi.com/Laptop/GT70_0NE/Specification
I use Win8.1 64 bit. Below is the generated DDrawCompat-Sims.log content after reinstalling 3.2.2, modifiying ini: FullscreenMode=borderless
CPUAffinity=all

and then running the game.

08:18:08.665 Process path: D:\Program Files\Maxis\The Sims\Sims.exe
08:18:08.665 Loading DDrawCompat dynamically from D:\Program Files\Maxis\The Sims\DDRAW.DLL
08:18:08.665 Environment variable __COMPAT_LAYER = "DWM8And16BitMitigation RunAsAdmin DetectorsXP"
08:18:08.665 Loading global config file: C:\ProgramData\DDrawCompat\DDrawCompat.ini
08:18:08.665   File not found, skipping
08:18:08.665 Loading user config file: C:\Users\Excolion\AppData\Local\DDrawCompat\DDrawCompat.ini
08:18:08.666   File not found, skipping
08:18:08.666 Loading directory config file: D:\Program Files\Maxis\The Sims\DDrawCompat.ini
08:18:08.666 Loading process config file: D:\Program Files\Maxis\The Sims\DDrawCompat-Sims.ini
08:18:08.666   File not found, skipping
08:18:08.666 Loading overlay config file: D:\Program Files\Maxis\The Sims\DDrawCompatOverlay-Sims.ini
08:18:08.666   File not found, skipping
08:18:08.666 Final configuration:
08:18:08.666   [default  ] AlignSysMemSurfaces   = on
08:18:08.666   [default  ] AltTabFix             = off
08:18:08.666   [default  ] AlternatePixelCenter  = off
08:18:08.666   [default  ] Antialiasing          = off
08:18:08.666   [default  ] BltFilter             = point
08:18:08.666   [default  ] ConfigHotKey          = shift+f11
08:18:08.666   [directory] CpuAffinity           = all
08:18:08.666   [default  ] DesktopColorDepth     = initial
08:18:08.666   [default  ] DisplayFilter         = bilinear(0)
08:18:08.666   [default  ] DisplayRefreshRate    = app
08:18:08.666   [default  ] DisplayResolution     = desktop
08:18:08.666   [default  ] DpiAwareness          = permonitor
08:18:08.666   [default  ] FontAntialiasing      = app
08:18:08.666   [default  ] ForceD3D9On12         = off
08:18:08.666   [default  ] FpsLimiter            = off
08:18:08.666   [directory] FullscreenMode        = borderless
08:18:08.666   [default  ] LogLevel              = info
08:18:08.666   [default  ] PalettizedTextures    = on
08:18:08.666   [default  ] RemoveBorders         = off
08:18:08.666   [default  ] RenderColorDepth      = app
08:18:08.666   [default  ] ResolutionScale       = app(1)
08:18:08.666   [default  ] ResolutionScaleFilter = point
08:18:08.666   [default  ] SoftwareDevice        = rgb
08:18:08.666   [default  ] SpriteDetection       = off
08:18:08.666   [default  ] SpriteFilter          = app
08:18:08.666   [default  ] SpriteTexCoord        = app
08:18:08.666   [default  ] SupportedResolutions  = native, 640x480, 800x600, 1024x768
08:18:08.666   [default  ] TextureFilter         = app
08:18:08.666   [default  ] ThreadPriorityBoost   = off
08:18:08.666   [default  ] VSync                 = app
08:18:08.666   [default  ] WinVersionLie         = off
08:18:08.675 DPI awareness was successfully changed to "permonitor" via SetProcessDpiAwareness
08:18:08.675 DDrawCompat loaded successfully
08:18:08.675 Installing display mode hooks
08:18:08.679 Installing registry hooks
08:18:08.679 Installing Direct3D driver hooks
08:18:08.684 Hooking user mode display driver: C:\WINDOWS\SYSTEM32\nvumdshim.dll+0x14d90
08:18:08.704 Dynamic vertex buffers are not available
08:18:08.704 Dynamic index buffers are not available
08:18:08.717 Supported z-buffer bit depths: 16, 24
08:18:08.717 Supported MSAA modes: msaa(0), msaa2x(0), msaa4x(0), msaa8x(0)
08:18:08.717 Supported resource formats:
08:18:08.717   {D3DDDIFMT_UNKNOWN,0x0,0x0,0x0,0}
08:18:08.717   {D3DDDIFMT_A8R8G8B8,0x408600f,0x8b,0x8b,0}
08:18:08.717   {D3DDDIFMT_X8R8G8B8,0x4086c0f,0x8b,0x8b,0}
08:18:08.717   {D3DDDIFMT_R5G6B5,0x86c0f,0x8b,0x8b,0}
08:18:08.717   {D3DDDIFMT_A1R5G5B5,0x8600f,0x8b,0x8b,0}
08:18:08.717   {D3DDDIFMT_A4R4G4B4,0x400f,0x8b,0x8b,0}
08:18:08.717   {D3DDDIFMT_A8,0x7,0x0,0x0,0}
08:18:08.717   {D3DDDIFMT_P8,0x4000,0x0,0x0,0}
08:18:08.717   {D3DDDIFMT_L8,0xf,0x0,0x0,0}
08:18:08.717   {D3DDDIFMT_A8L8,0x7,0x0,0x0,0}
08:18:08.717   {D3DDDIFMT_A4L4,0x7,0x0,0x0,0}
08:18:08.717   {D3DDDIFMT_V8U8,0x7,0x0,0x0,0}
08:18:08.717   {D3DDDIFMT_L6V5U5,0x7,0x0,0x0,0}
08:18:08.717   {D3DDDIFMT_Q8W8V8U8,0x7,0x0,0x0,0}
08:18:08.717   {D3DDDIFMT_S8D24,0xc1,0x8b,0x8b,0}
08:18:08.717   {D3DDDIFMT_X8D24,0xc1,0x8b,0x8b,0}
08:18:08.717   {D3DDDIFMT_D16,0xc1,0x8b,0x8b,0}
08:18:08.717   {FOURCC(I420),0x6000,0x0,0x0,0}
08:18:08.717   {FOURCC(NV11),0x7000,0x0,0x0,0}
08:18:08.717   {FOURCC(IMC1),0x6000,0x0,0x0,0}
08:18:08.717   {FOURCC(DXT1),0x7,0x0,0x0,0}
08:18:08.717   {FOURCC(NV12),0x6000,0x0,0x0,0}
08:18:08.717   {FOURCC(YV12),0x6000,0x0,0x0,0}
08:18:08.717   {FOURCC(IMC2),0x6000,0x0,0x0,0}
08:18:08.717   {FOURCC(DXT2),0x7,0x0,0x0,0}
08:18:08.717   {FOURCC(YUY2),0x4006001,0x0,0x0,0}
08:18:08.717   {FOURCC(IMC3),0x6000,0x0,0x0,0}
08:18:08.717   {FOURCC(DXT3),0x7,0x0,0x0,0}
08:18:08.717   {FOURCC(IA44),0x4000,0x0,0x0,0}
08:18:08.717   {FOURCC(AI44),0x4000,0x0,0x0,0}
08:18:08.717   {FOURCC(IMC4),0x6000,0x0,0x0,0}
08:18:08.717   {FOURCC(DXT4),0x7,0x0,0x0,0}
08:18:08.717   {FOURCC(DXT5),0x7,0x0,0x0,0}
08:18:08.717   {FOURCC(P208),0x6000,0x0,0x0,0}
08:18:08.717   {FOURCC(YVU9),0x6000,0x0,0x0,0}
08:18:08.717   {FOURCC(YVYU),0x4006000,0x0,0x0,0}
08:18:08.717   {FOURCC(AYUV),0x4009,0x0,0x0,32}
08:18:08.717   {FOURCC(IYUV),0x6000,0x0,0x0,0}
08:18:08.717   {FOURCC(VYUY),0x4006000,0x0,0x0,0}
08:18:08.717   {FOURCC(UYVY),0x4006009,0x0,0x0,16}
08:18:08.720 Installing DirectDraw hooks
08:18:08.721 Installing Direct3D hooks
08:18:08.724 Installing GDI hooks
08:18:08.732 Finished installing hooks
08:18:13.046 Using Direct3D device type: IID_IDirect3DRGBDevice

**Comment by voicemxil** [comment]
The log doesn't seem to be that helpful bc it doesn't explain the crash :/. Did TS1 itself make any kind of logs?

It's possible that it's an 8.1 issue (may test in a VM) or something setup related, also I still need to look at your specs later when I get a chance

**Comment by Gaerthal-Estelmist** [comment]
just tried it, didn't make a difference.

**Comment by Akoi89** [comment]
I am experiencing the same issue on windows 7, I thought it was just windows 8.1 yesterday so I reinstalled windows 7 on this pc and I am getting the same issue.

**Comment by voicemxil** [comment]
> I am experiencing the same issue on windows 7, I thought it was just windows 8.1 yesterday so I reinstalled windows 7 on this pc and I am getting the same issue.
> 
> The file from: https://github.com/narzoul/DDrawCompat/files/9736501/ddraw.zip seems to fix it tho for windows 7. (never tested this file on windows 8.1)

Interesting choice to upgrade to an even older version lol. Sadly that file did not fix the problem for 8.1.

**Comment by Akoi89** [comment]
> 

what I did notice is natively at 1024x768 this game under windows 8.1 runs around 10-20fps, and under windows 7 it sticks around 60fps... might just be my hardware but that was an odd thing I discovered as well.

**Comment by FaithBeam** [comment]
Has 0.4.0 worked at all for Windows 7 and Windows 8.1 users for Sims 1? If not, it's easy enough to detect the version of Windows at runtime and change the version of ddraw based on that.

**Comment by voicemxil** [comment]
> Has 0.4.0 worked at all for Windows 7 and Windows 8.1 users for Sims 1? If not, it's easy enough to detect the version of Windows at runtime and change the version of ddrawcompat based on that.

I've been testing with narzoul, and that fix @Akoi89 mentioned did end up working for 7 and 8.1. My VM's resolution was set too low when I sent the last message here, which caused an additional crash. 
Hopefully it'll be implemented in the next main release, but for now this version of 0.4.0 fixes the 7/8 issues including crash when debug logging and the crash to desktop. It also works on later Windows versions with no issues. https://github.com/narzoul/DDrawCompat/issues/217?notification_referrer_id=NT_kwDOBB3FFrM1OTI5OTkzMzY5OjY5MDU5ODYy#issuecomment-1481733147

**Comment by Gaerthal-Estelmist** [comment]
> Has 0.4.0 worked at all for Windows 7 and Windows 8.1 users for Sims 1? If not, it's easy enough to detect the version of Windows at runtime and change the version of ddrawcompat based on that.

Hi, your v 3.3.0 did not work for me, so I went back to the last stable one, which is v 3.2.3 .

**Comment by voicemxil** [comment]
> > Has 0.4.0 worked at all for Windows 7 and Windows 8.1 users for Sims 1? If not, it's easy enough to detect the version of Windows at runtime and change the version of ddrawcompat based on that.
> 
> 
> 
> Hi, your v 3.3.0 did not work for me, so I went back to the last stable one, which is v 3.2.3 .

We know the normal version of 0.4.0 doesn't work, did you skip past the last messages? narzoul has made a fixed version of 0.4.0 for windows 7/8 that i linked.

**Comment by Gaerthal-Estelmist** [comment]
> > Hi, your v 3.3.0 did not work for me, so I went back to the last stable one, which is v 3.2.3 .
> 
> @Gaerthal-Estelmist did you skip past the last message? narzoul has made a fixed version of ddrawcompat 0.4.0 for windows 7/8, @Akoi89 and I have linked it.

I should have given more details.
I installed v 3.3.0, replaced ddraw.dll with the one in your comment. The game worked but it was too laggy. Both the mouse and screen scrolling feels like walking through the mud. This is why it is not working for me. At least the CTD is fixed, which is good news but I'll have to continue using 3.2.3 for the time being.
